backend/api/system/applications.py

```diff
@@ -1,132 +1,155 @@
+    """
+    tokens = await ApplicationService.get_all_tokens_for_logged_in_user(user.id, db)
+    return tokens.model_dump(by_alias=True)
 
-    def patch(self, application_key):
-        """
-        Checks the validity of an application key
-        ---
-        tags:
-          - system
-        produces:
-          - application/json
-        parameters:
-          - in: path
-            name: application_key
-            description: Application key to test
-            type: string
-            required: true
-            default: 1
-        responses:
-          200:
-            description: Key is valid
-          302:
-            description: Key is not valid
-          500:
-            description: A problem occurred
-        """
-        is_valid = ApplicationService.check_token(application_key)
-        if is_valid:
-            return 200
-        else:
-            return 302
 
-    @token_auth.login_required
-    def delete(self, application_key):
-        """
-        Deletes an application key for a user
-        ---
-        tags:
-          - system
-        produces:
-          - application/json
-        parameters:
-          - in: header
-            name: Authorization
-            description: Base64 encoded session token
-            required: true
-            type: string
-            default: Token sessionTokenHere==
-          - in: path
-            name: application_key
-            description: Application key to remove
-            type: string
-            required: true
-            default: 1
-        responses:
-          200:
-            description: Key deleted successfully
-          302:
-            description: User is not authorized to delete the key
-          404:
-            description: Key not found
-          500:
-            description: A problem occurred
-        """
-        token = ApplicationService.get_token(application_key)
-        if token.user == token_auth.current_user():
-            token.delete()
-            return 200
-        else:
-            return 302
+@router.post("/authentication/applications/")
+async def post(
+    request: Request,
+    user: AuthUserDTO = Depends(login_required),
+    db: Database = Depends(get_db),
+):
+    """
+    Creates an application key for the user
+    ---
+    tags:
+      - system
+    produces:
+      - application/json
+    parameters:
+      - in: header
+        name: Authorization
+        description: Base64 encoded session token
+        required: true
+        type: string
+        default: Token sessionTokenHere==
+    responses:
+      200:
+        description: Key generated successfully
+      302:
+        description: User is not authorized to create a key
+      500:
+        description: A problem occurred
+    """
+    token = await ApplicationService.create_token(user.id, db)
+    return token.model_dump(by_alias=True)
+
+
+@router.patch("/authentication/applications/{application_key}/")
```

Comment:
Using a PATCH method here seems a bit out of place, but maybe I am missing something